library/network/src/modules/PortAliases.rb

```diff
@@ -141,175 +97,143 @@ def AllowedPortNameOrNumber
-          LoadAndReturnPortToName(port_number)
-        end
+      # Then, process those returned by `getent servies`
+      load_services_database.each do |service|
+        name, port, aliases = service.chomp.split(" ")
 
-        if !sport_name.nil?
-          service_aliases = Convert.convert(
-            Builtins.union(
-              service_aliases,
-              Builtins.splitstring(sport_name, " ")
-            ),
-            from: "list",
-            to:   "list <string>"
-          )
-        end
-        # service is a port name, any space isn't allowed
-      elsif IsAllowedPortName(port)
-        found_alias_port = Ops.get(@SERVICE_NAME_TO_PORT, port) do
-          LoadAndReturnNameToPort(port)
-        end
-        if !found_alias_port.nil?
-          service_aliases = Builtins.add(
-            service_aliases,
-            Builtins.tostring(found_alias_port)
-          )
+        port = port.to_i
+        aliases = [name, aliases&.split].flatten.compact
+        service = find_by_port(port)
 
-          # search for another port-name aliases when port-number found
-          service_aliases = Convert.convert(
-            Builtins.union(
-              service_aliases,
-              Builtins.splitstring(
-                Ops.get(@SERVICE_PORT_TO_NAME, found_alias_port) do
-                  LoadAndReturnPortToName(found_alias_port)
-                end,
-                " "
-              )
-            ),
-            from: "list",
-            to:   "list <string>"
-          )
-        end
-      else
-        if !Builtins.contains(@cache_not_allowed_ports, port)
-          @cache_not_allowed_ports = Builtins.add(
-            @cache_not_allowed_ports,
-            port
-          )
-          Builtins.y2error("Port name '%1' is not allowed", port)
+        if service
+          service.aliases |= aliases
         else
-          Builtins.y2debug("Port name '%1' is not allowed", port)
+          service = Service.new(port, aliases)
+          @services << service
         end
-        return [port]
       end
 
-      Builtins.toset(service_aliases)
+      @services
     end
 
-    # Function returns if the requested port-name is known port.
-    # Known port have an IANA alias.
+    # Returns services database after performing some cleanup
     #
-    # @param  string port-name
-    # @return  [Boolean] if is known
-    def IsKnownPortName(port_name)
-      # function returns the requested port and aliases if exists
-      return true if Ops.greater_than(Builtins.size(GetListOfServiceAliases(port_name)), 1)
-
-      false
+    # Basically, it manipulates the `getent services` output to discard duplicated lines after
+    # removing the protocol.
+    #
+    # Returned lines will look like
+    #
+    # EtherNet/IP-1         2222
+    # EtherNet-IP-2         44818
+    # rfb                   5900 vnc-server
+    #
+    # @return [Array<String>]
+    def load_services_database
```

Comment:
Why not just `services_database` (or something similar)? The `load_` part suggests that the information is being "loaded/cached/stored" in some place.

Reply:
In fact, it's being _loaded_ via `getent`